### ISSUE TITLE
Feature Request: Allow include/match-specific callback function. 

### ISSUE THREAD

**Issue by zanetu** [greasemonkey/greasemonkey#2075]
TL;DR: Allow a callback function to be added to each include/match rule. 

Long version: 
Some popular userscripts operate on a multitude of websites. (e.g. https://greasyfork.org/en/scripts/2600,  https://greasyfork.org/en/scripts/4294, and https://greasyfork.org/en/scripts/293) Since they have to handle a good many heterogeneous websites, they usually have a lengthy part of code that determines which website the userscript is currently working at. 

This is not only tedious but also time-wasting, as the metablock has already ascertained which website the userscript is working at. 

Why not add a callback (string) parameter to each include/match rule (or create new imperatives such as @include-callback and @match-callback) so that the userscript immediately knows how to proceed after the metablock is parsed? 

The implementation would not be very complicated because it is almost equivalent to appending a line of code (depending on which include/match rule serves as the "entry point") to the end of the original userscript. 

Short Example: 

```
// @inlucde http://www.google.com/ "doGoogle('e')"
// @inlucde http://www.yahoo.com/ "doYahoo('Yahoo!')"
function doGoogle(lastLetter) {
    console.log("Last letter: " + lastLetter)
}
function doYahoo(yell) {
    alert(yell)
}
//Either "doGoogle('e')" or "doYahoo('Yahoo!')" is appended here. 
```

Long Example: 
https://greasyfork.org/en/scripts/566 -> http://www.hastebin.com/ogovuzijun.js
Benefits: 
1. Lines of code: 184 -> 89
2. File size: 5.4KB -> 3.7KB
3. Efficiency: Improved
4. This also has the side effect of encouraging "lazy" userscript authors to avoid applying their userscripts to all websites on the Internet. (e.g. https://greasyfork.org/en/scripts/2600)
